mmcv/runner/hooks/lr_updater.py

```diff
@@ -473,21 +482,27 @@ def before_run(self, runner):
         super(CyclicLrUpdaterHook, self).before_run(runner)
         # initiate lr_phases
         # total lr_phases are separated as up and down
-        max_iter_per_phase = runner.max_iters // self.cyclic_times
-        iter_up_phase = int(self.step_ratio_up * max_iter_per_phase)
-        self.lr_phases.append(
-            [0, iter_up_phase, max_iter_per_phase, 1, self.target_ratio[0]])
+        self.max_iter_per_phase = runner.max_iters // self.cyclic_times
+        iter_up_phase = int(self.step_ratio_up * self.max_iter_per_phase)
+        self.lr_phases.append([0, iter_up_phase, 1, self.target_ratio[0]])
         self.lr_phases.append([
-            iter_up_phase, max_iter_per_phase, max_iter_per_phase,
-            self.target_ratio[0], self.target_ratio[1]
+            iter_up_phase, self.max_iter_per_phase, self.target_ratio[0],
+            self.target_ratio[1]
         ])
 
     def get_lr(self, runner, base_lr):
-        curr_iter = runner.iter
-        for (start_iter, end_iter, max_iter_per_phase, start_ratio,
-             end_ratio) in self.lr_phases:
-            curr_iter %= max_iter_per_phase
+        curr_iter = runner.iter % self.max_iter_per_phase
+        curr_cycle = runner.iter // self.max_iter_per_phase
+        # Update weight decay
+        scale = self.gamma**curr_cycle
+
+        for (start_iter, end_iter, start_ratio, end_ratio) in self.lr_phases:
             if start_iter <= curr_iter < end_iter:
+                # Apply cycle scaling to gradually reduce max_lr.
+                if end_ratio > start_ratio:
+                    end_ratio = 1 - scale + end_ratio * scale
+                else:
+                    start_ratio = 1 - scale + start_ratio * scale
```

Comment:
We need to add a comment to describe why the end_ratio will be `1 - scale + end_ratio * scale` like `To ensure that peak_lr is larger than base_lr, we add 1-scale as a bias. If self.target_ratio is greater than 1, this ensures that end_ratio or start_ratio is always greater than 1, i.e. peak_lr is greater than base_lr.` In addition, we should add more description in the pr message with several figures.